pkg/schema/converter/crd_test.go

```diff
@@ -0,0 +1,459 @@
+package converter
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/rancher/apiserver/pkg/types"
+	"github.com/rancher/steve/pkg/schema/table"
+	wranglerSchema "github.com/rancher/wrangler/pkg/schemas"
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	kubeTypes "k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+func TestAddCustomResources(t *testing.T) {
+	tests := []struct {
+		name             string
+		crds             []v1.CustomResourceDefinition
+		preFilledSchemas []string // crds to pre-create schemas for
+		crdError         error
+		wantError        bool
+		desiredSchema    map[string]*types.APISchema
+	}{
+		{
+			name: "one crd - all different field types",
+			crds: []v1.CustomResourceDefinition{
+				{
+					Status: v1.CustomResourceDefinitionStatus{
+						AcceptedNames: v1.CustomResourceDefinitionNames{
+							Plural:   "testResources",
+							Singular: "testResource",
+							Kind:     "TestResource",
+						},
+					},
+					Spec: v1.CustomResourceDefinitionSpec{
+						Group: "testGroup",
+						Versions: []v1.CustomResourceDefinitionVersion{
+							{
+								Name: "v1",
+								AdditionalPrinterColumns: []v1.CustomResourceColumnDefinition{
+									{
+										Name:     "TestColumn",
+										JSONPath: "TestPath",
+										Type:     "TestType",
+										Format:   "TestFormat",
+									},
+								},
+								Schema: &v1.CustomResourceValidation{
+									OpenAPIV3Schema: &v1.JSONSchemaProps{
+										Description: "Test Resource for unit tests",
+										Required:    []string{"required"},
+										Properties: map[string]v1.JSONSchemaProps{
```

Comment:
Are all these properties required for testing, or could they be condensed into one or two for the sake of readability?

Reply:
I believe that each of these covers a test case for [toResourceField](https://github.com/rancher/steve/blob/1061dd34560d66e31d1d028d7278a6efb6d251c0/pkg/schema/converter/openapiv3.go#L45). I'd like to keep them in here since it asserts on the behavior we expect from all of our types. I could try and restructure these test cases so that there's one that does assertions only on properties, but I think that's a fair amount of changes for test readability. 

So in short, I do think that these are required for full test coverage. I can try and refactor them if you think it's worth it.